components/brave_wallet/common/brave_wallet.mojom

```diff
@@ -942,6 +942,9 @@ interface JsonRpcService {
   // SNS lookups
   SnsGetSolAddr(string domain) => (string address, SolanaProviderError error, string error_message);
 
+  // SNS Domain lookups
+  SnsResolveHost(string domain) => (url.mojom.Url? url, SolanaProviderError error, string error_message);
```

Comment:
Since I pushed commits myself, I delegate the review to original author @supermassive.
iOS needs it to be optional to work because 
>our native URL type NSURL cannot be represented by an invalid URL like GURL. To be specific, the converter method we use returns nil when the GURL is invalid: https://source.chromium.org/chromium/chromium/src/+/main:net/base/mac/url_conversions.mm;l=18

Quoted from @kylehickinson 

And it should have been declared as optional for those trivial early errors instead of using invalid GURL